conf.py

```diff
@@ -48,30 +65,54 @@
 nb_output_stderr = "remove"
 myst_heading_anchors = 3
 
-render_image_orig = NbElementRenderer.render_image
+
+# Roles “implementing” {cite}`…` and {cite:p}`…`/{cite:t}`…`
+
+
+def fake_cite(
+    name: str,
+    rawtext: str,
+    text: str,
+    lineno: int,
+    inliner: Inliner,
+    options: Mapping[str, object] = MappingProxyType({}),
+    content: Sequence[str] = (),
+) -> tuple[list[nodes.Node], list[str]]:
+    node = inliner.document.reporter.info(f"cite:{text}")
+    return [node], []
+
+
+class FakeDomain(Domain):
+    name = "cite"
+    roles = dict(p=fake_cite, t=fake_cite)
+
```

Comment:
Is this for the warnings?  What does it do?

Reply:
What it says in the comment in line 69: It makes ``{cite}`Foo17` `` and ``{cite:p}`Bar22` `` not crash the docs build.

Reply:
But good point, I changed it so it actually shows the citation text